metadata-ingestion/src/datahub_provider/client/dagster_generator.py

```diff
@@ -0,0 +1,310 @@
+import os
+from typing import Dict, List, Optional
+
+import pydantic
+from dagster._core.execution.stats import RunStepKeyStatsSnapshot, StepEventStatus
+from dagster._core.snap import JobSnapshot
+from dagster._core.snap.node import OpDefSnap
+from dagster._core.storage.dagster_run import (
+    DagsterRun,
+    DagsterRunStatsSnapshot,
+    DagsterRunStatus,
+)
+
+from datahub.api.entities.datajob import DataFlow, DataJob
+from datahub.api.entities.dataprocess.dataprocess_instance import (
+    DataProcessInstance,
+    InstanceRunResult,
+)
+from datahub.configuration.source_common import DatasetSourceConfigMixin
+from datahub.emitter.rest_emitter import DatahubRestEmitter
+from datahub.utilities.urns.data_flow_urn import DataFlowUrn
+from datahub.utilities.urns.data_job_urn import DataJobUrn
+from datahub.utilities.urns.dataset_urn import DatasetUrn
+
+
+class Constant:
+    """
+    keys used in dagster plugin
+    """
+
+    ORCHESTRATOR = "dagster"
+
+    # Default config constants
+    DEFAULT_DATAHUB_REST_URL = "http://localhost:8080"
+
+    # Environment variable contants
+    DATAHUB_REST_URL = "DATAHUB_REST_URL"
+    DATAHUB_ENV = "DATAHUB_ENV"
+    DATAHUB_PLATFORM_INSTANCE = "DATAHUB_PLATFORM_INSTANCE"
+
+    # Datahub inputs/outputs constant
+    DATAHUB_INPUTS = "datahub.inputs"
+    DATAHUB_OUTPUTS = "datahub.outputs"
+
+    # Job run constant
+    JOB_SNAPSHOT_ID = "job_snapshot_id"
+    EXECUTION_PLAN_SNAPSHOT_ID = "execution_plan_snapshot_id"
+    ROOT_RUN_ID = "root_run_id"
+    PARENT_RUN_ID = "parent_run_id"
+    HAS_REPOSITORY_LOAD_DATA = "has_repository_load_data"
+    TAGS = "tags"
+    STEPS_SUCCEEDED = "steps_succeeded"
+    STEPS_FAILED = "steps_failed"
+    MATERIALIZATIONS = "materializations"
+    EXPECTATIONS = "expectations"
+    ENQUEUED_TIME = "enqueued_time"
+    LAUNCH_TIME = "launch_time"
+    START_TIME = "start_time"
+    END_TIME = "end_time"
+
+    # Op run contants
+    STEP_KEY = "step_key"
+    ATTEMPTS = "attempts"
+
+
+class DagsterSourceConfig(DatasetSourceConfigMixin):
+    datahub_rest_url: str = pydantic.Field(
+        default=Constant.DEFAULT_DATAHUB_REST_URL,
+        description="Datahub GMS Rest URL. Example: http://localhost:8080",
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if Constant.DATAHUB_REST_URL in os.environ:
+            self.datahub_rest_url: str = os.environ[Constant.DATAHUB_REST_URL]
```

Comment:
Please change this to os.getenv(&lt;env-name&gt;, &lt;default-value&gt;), It will remove all if condition